### PR TITLE
Issue1.2 #1

### PR DESCRIPTION
**Pull request by yoona3316**
Change bottom nav bar to have 3 buttons.

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1,3 +1,6 @@
 <resources>
     <string name="app_name">MCATeam6</string>
+    <string name="title_search">Search</string>
+    <string name="title_barcode">Barcode</string>
+    <string name="title_setting">Setting</string>
```
Comment (PhilippMa) [inline app/src/main/res/values/strings.xml]: Should be "Settings"

File: app/src/main/res/menu/bottom_nav_menu.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <item
+        android:id="@+id/navigation_search"
+        android:icon="@drawable/ic_home_black_24dp"
```
Comment (PhilippMa) [inline app/src/main/res/menu/bottom_nav_menu.xml]: We can use some better fitting icons, search/camera/settings icon are all available in res -> New -> Vector Asset

File: app/src/main/java/com/example/mcateam6/fragments/BarcodeFragment.kt [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import com.example.mcateam6.activities.MainActivity
+
+private const val REQUEST_CODE_PERMISSIONS = 10
+
+private val REQUIRED_PERMISSIONS = arrayOf(Manifest.permission.CAMERA)
+
+class BarcodeFragment : Fragment() {
+    private lateinit var view_finder: TextureView
```
Comment (PhilippMa) [inline app/src/main/java/com/example/mcateam6/fragments/BarcodeFragment.kt]: Not that important, but Android Studio complains about naming conventions (view_finder instead of viewFinder)

**Review by PhilippMa — COMMENTED** [review]
In my opinion we don't really need the action bar and could just change our theme to inherit from "Theme.MaterialComponents.Light.NoActionBar". Also, I found an easy way to make the activity go fullscreen with a transparent status bar: https://proandroiddev.com/android-full-screen-ui-with-transparent-status-bar-ef52f3adde63

**Comment by yoona3316** [comment]
sorry for being late. Just finished fixing the resolutions.

File: app/src/main/res/values/styles.xml [hunk 4]
```diff
@@ -1,11 +1,10 @@
 <resources>
 
     <!-- Base application theme. -->
-    <style name="AppTheme" parent="Theme.AppCompat.Light.DarkActionBar">
+    <style name="AppTheme" parent="Theme.AppCompat.Light.NoActionBar">
```
Comment (PhilippMa) [inline app/src/main/res/values/styles.xml]: Can we please use "Theme.MaterialComponents.Light.NoActionBar" instead? I need that to use the chips in my branch

File: app/src/main/res/menu/bottom_nav_menu.xml [hunk 5]
```diff
@@ -3,17 +3,17 @@
...
     <item
         android:id="@+id/navigation_barcode"
-        android:icon="@drawable/ic_dashboard_black_24dp"
+        android:icon="@drawable/ic_camera_alt_24px"
         android:title="@string/title_barcode" />
 
     <item
         android:id="@+id/navigation_setting"
-        android:icon="@drawable/ic_notifications_black_24dp"
+        android:icon="@drawable/ic_settings_applications_24px"
```
Comment (PhilippMa) [inline app/src/main/res/menu/bottom_nav_menu.xml]: I think the "settings" icon provided by android studio looks better than the "settings_application" icon

File: app/src/main/res/layout/activity_main.xml [hunk 6]
```diff
@@ -12,11 +12,13 @@
         android:layout_height="wrap_content"
         android:layout_marginStart="0dp"
         android:layout_marginEnd="0dp"
-        android:background="?android:attr/windowBackground"
+        android:background="@color/colorPrimary"
```
Comment (PhilippMa) [inline app/src/main/res/layout/activity_main.xml]: With material theme, you can revert the changes to this file and get a white action bar with green highlighted icons and text

File: app/src/main/java/com/example/mcateam6/activities/MainActivity.kt [hunk 7]
```diff
@@ -24,7 +24,6 @@ class MainActivity : AppCompatActivity() {
                 R.id.navigation_search, R.id.navigation_barcode, R.id.navigation_setting
             )
         )
-        setupActionBarWithNavController(navController, appBarConfiguration)
```
Comment (PhilippMa) [inline app/src/main/java/com/example/mcateam6/activities/MainActivity.kt]: Please also delete val appBarConfiguration since it is no longer used

File: app/src/main/java/com/example/mcateam6/activities/MainActivity.kt [hunk 8]
```diff
@@ -24,7 +24,6 @@ class MainActivity : AppCompatActivity() {
...
         navView.setupWithNavController(navController)
     }
```
Comment (PhilippMa) [inline app/src/main/java/com/example/mcateam6/activities/MainActivity.kt:30]: Add this code snippet in onCreate to make the app go fullscreen:

```kotlin
window.apply {
            clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS)
            addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS)
            decorView.systemUiVisibility = View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN or View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR
            statusBarColor = Color.TRANSPARENT
        }
```

File: app/src/main/AndroidManifest.xml [hunk 9]
```diff
@@ -12,6 +12,7 @@
         android:theme="@style/AppTheme">
         <activity
             android:name=".activities.MainActivity"
+            android:theme="@style/Theme.AppCompat.Light.NoActionBar"
```
Comment (PhilippMa) [inline app/src/main/AndroidManifest.xml]: Why does the activity need its own theme declaration when it uses the same theme as the app? I think you can just remove this line
